src/Connection.h

```diff
@@ -32,6 +33,20 @@ class Message;
 }
 }
 
+enum VoiceProtocolType {UNDEFINED, CUSTOM, UDP_OCB2};
+struct VoiceProtocol {
+	VoiceProtocolType protocolType;
+	std::string customProtocolName;
+	explicit VoiceProtocol (VoiceProtocolType t) {
+		protocolType = t;
+		customProtocolName = "";
+	}
+	explicit VoiceProtocol (std::string s) {
+		protocolType = VoiceProtocolType::CUSTOM;
+		customProtocolName = std::move(s);
```

Comment:
Clang-Tidy warns me to use std::move to avoid copying.

Reply:
You should instead not pass the string by value into the constructor. Pass it by const reference and then copy it in side the constructor.
Additionally you can implement a move-constructor that'll take an rvalue reference and then moves the given string. But I don't think it is necessary to do so as this is not an object that gets constructed a lot.

Reply:
Got it.